### DISCUSSION TITLE
OpenAI assistant - predominantly Code Interpreter as SK function (C#)

### DISCUSSION THREAD

**Discussion by JP-droidic** [microsoft/semantic-kernel#7237] [Q&A] [answered]
Do you have a suggestion on how to mix "normal" SK with OpenAI assistant (predominantly Code Interpreter). We built all our flows around SK and auto function invoke, but we want to add some advanced things like creating a graph from the data. 

In particular, my question is, should we wrap an OpenAI assistant call as SK function with a description like - if you do not have the skills to answer the user's request, call this function (and hope this function will be auto invoked will ) or do you have any ther alternative.

Basically, we want to allow our users some out-of-the-box functionalities and the code interpreter sounds like the perfect fit. But all our code base is based on non-agent flow/threads.

**Comment by sophialagerkranspandey** [comment, marked as answer]
[semantic-kernel/dotnet/samples/Demos/CodeInterpreterPlugin at main · microsoft/semantic-kernel (github.com)](https://github.com/microsoft/semantic-kernel/tree/main/dotnet/samples/Demos/CodeInterpreterPlugin)
Here's a sample that showcases how you don't need to use the OpenAI assistant to get access to the code interpreter - here we show how to use Azure Container apps to get the same thing.
